warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -1096,20 +1096,47 @@ private Range getDocumentRange(NestedQueryIterator<Key> documentSource) {
     }
     
     protected JexlEvaluation getJexlEvaluation(NestedQueryIterator<Key> documentSource) {
+        return getJexlEvaluation(query, documentSource, getArithmetic());
+    }
+    
+    protected JexlEvaluation getJexlEvaluation(String query) {
+        return getJexlEvaluation(query, null, getArithmetic());
+    }
+    
+    protected JexlEvaluation getJexlEvaluation(String query, JexlArithmetic arithmetic) {
+        return getJexlEvaluation(query, null, arithmetic);
+    }
+    
+    protected JexlEvaluation getJexlEvaluation(String query, NestedQueryIterator<Key> documentSource) {
+        return getJexlEvaluation(query, documentSource, getArithmetic());
+    }
+    
+    protected JexlEvaluation getJexlEvaluation(String query, NestedQueryIterator<Key> documentSource, JexlArithmetic arithmetic) {
+        JexlEvaluation jexlEvaluationFunction = null;
         
-        if (null == documentSource) {
-            return new JexlEvaluation(query, getArithmetic());
+        if (arithmetic == null) {
+            arithmetic = getArithmetic();
         }
-        JexlEvaluation jexlEvaluationFunction = null;
-        NestedQuery<Key> nestedQuery = documentSource.getNestedQuery();
-        if (null == nestedQuery) {
-            jexlEvaluationFunction = new JexlEvaluation(query, getArithmetic());
+        
+        if (null == documentSource) {
+            jexlEvaluationFunction = new JexlEvaluation(query, arithmetic);
         } else {
-            jexlEvaluationFunction = nestedQuery.getEvaluation();
-            if (null == jexlEvaluationFunction) {
-                return new JexlEvaluation(query, getArithmetic());
+            NestedQuery<Key> nestedQuery = documentSource.getNestedQuery();
+            if (null == nestedQuery) {
+                jexlEvaluationFunction = new JexlEvaluation(query, arithmetic);
+            } else {
+                jexlEvaluationFunction = nestedQuery.getEvaluation();
+                if (null == jexlEvaluationFunction) {
+                    jexlEvaluationFunction = new JexlEvaluation(query, arithmetic);
+                }
             }
         }
+        
+        // update the jexl evaluation to gather phrase offsets if required for excerpts
+        if (getExcerptFields() != null && !getExcerptFields().isEmpty()) {
+            jexlEvaluationFunction.setGatherPhraseOffsets(true);
+        }
```

Comment:
Might want to create an issue to diff the excerpt fields against the fields in the query's content function. It's possibly (though unlikely) that the content function fields in the query do not match the fields specified in the excerpt fields -- as the result of query model expansion and/or range stream pruning.

Reply:
#1607